Root installation, update dependencies & more default repositories

machiav3lli opened the pull request:
This PR include:
- Updating the dependencies, Kotlin, Gradle and AGP
- Raising target- and compileSDK to 31 and add needed permissions
- Add root installation method
- Add 23 default repository
- True black as background in dark theme

machiav3lli committed:
Update: Gradle wrapper

machiav3lli committed:
Update: Dependencies

machiav3lli committed:
Update: AGP

machiav3lli committed:
Update: Tweak gradle build

machiav3lli committed:
Update: Kotlin

machiav3lli committed:
Update: Migrate from Jcenter to MavenCentral

machiav3lli committed:
Update: Compile- and targetSDK to 31

machiav3lli committed:
Fix: Removed/deprecated Kotlin functions

machiav3lli committed:
Add: Root installation preference

machiav3lli committed:
Add: Root installation

machiav3lli committed:
Update: True/amoled black dark mode

machiav3lli committed:
Add: Query all apps permission (needed for SDK30+)

machiav3lli committed:
Fix: Root installation key name

machiav3lli committed:
Update: Grey out root installation option when not available

machiav3lli committed:
Update: Allow root installation in background even when in other Activities as ScreenActivity

machiav3lli committed:
Update: Make startPackageInstaller an extension of Activity. Add: Clean up cached apk after install

machiav3lli committed:
Add: 23 repositories to default

machiav3lli committed:
Update: Sort default repositories by name

Madis0 commented:
Are you sure the owner agrees with all your additions and it is easy to discern them? It might be a better idea to split the features into different PRs

machiav3lli commented:
> Are you sure the owner agrees with all your additions and it is easy to discern them? It might be a better idea to split the features into different PRs

In #102 I try to explain my position to the maintaining of the project. Eventually I'd consider those changes as a first step in direction co-maintaining the project. If such is not wished/wanted I'd be starting an independent fork.

RickyM7 commented:
@machiav3lli If you are really going to continue this project in a new fork, please talk here, as I want to follow this project, after all Foxy Droid is the best F-Droid client in my opinion.

ItsHarper commented:
Note that adding a bunch of new repositories to the `defaultRepositories` list is not good enough to make them available to existing users, because that list only gets referenced when the database is first created (not just updated):

https://github.com/kitsunyan/foxy-droid/blob/d7b3476a4a2aa5fc654065d30385c857de3963b2/src/main/kotlin/nya/kitsunyan/foxydroid/database/Database.kt#L24-L27

New code needs to be added that is capable of adding default repositories only if they aren't already in the database (including if they are in the database, but marked as deleted).

BTW, I'm the developer of [a sort of Foxy Droid fork myself](https://github.com/REVrobotics/Driver-Hub-Software-Manager). While the Driver Hub Software Manager has very different goals from Foxy Droid, I have implemented some functionality (such as updating all apps with a single button press) that could be useful to Foxy Droid users.

3504k commented:
For the love of FOSS and all things holy in this world, I pray in the name of Izzy to all the open source gods to please make a usable fork of this project.

machiav3lli committed:
Update: Don't open app's page after installing/updating

machiav3lli committed:
Update: Move toInstalledItem and abstract install/update action to Utils

machiav3lli committed:
Update: Update icon

machiav3lli committed:
Add: Update all (skeleton)

The codebase has some structural design problems and so needs some huge refactoring.

another-sapiens commented:
@3504k Droid-ify!

machiav3lli committed:
Update: Hide update all button else where as updates page

machiav3lli committed:
Fix: Installing new app through install button
